packages/shared/src/utils/platform/platform.js

```diff
@@ -128,3 +137,13 @@ export const isNavigationFromP2P = (routing_history, platform_route) => {
     const history_item = routing_history[routing_history_index];
     return history_item?.pathname === platform_route;
 };
+
+export const isNavigationFromExternalPlatform = (routing_history, platform_route) => {
+    for (let i = 0; i < routing_history.length; i++) {
+        const history_item = routing_history[i];
+
+        if (history_item.pathname === platform_route) return true;
+    }
+
+    return false;
```

Comment:
I think you can write this function with a simpler logic

```suggestion
export const isNavigationFromExternalPlatform = (routing_history, platform_route) => {
    return routing_history.some(history_item => history_item.pathname === platform_route);
}
```